### PR TITLE
Docs: Editing the CONTRIBUTING.md file

### PR DESCRIPTION
**Pull request by kirszenbaum**
This PR:
- Adds info about adding a mandatory changelog entry
- Makes it absolutely clear that build files should not be committed
- Rewrites the content, a bit

[skip changelog]

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,31 +1,42 @@
...
-4. Always make the target of your pull request the `develop` branch, not `master`.
-5. Do not edit files in the following directories:
+To speed up the process of merging your changes, follow these rules:
+
+1. Sign the [Contributor License Agreement](https://goo.gl/forms/yuutGuN0RjsikVpM2), to let us publish your changes.
+2. Make your changes on a separate branch. This will speed up the merging process.
+3. Always target your PR at the `develop` branch, not the `master` branch.
+4. Make changes to the right project:
+    - The main Handsontable project is located in the `./handsontable/` directory.
+    - Framework wrapper projects are located in the `./wrappers/` directory.
```
Comment (budnix) [inline CONTRIBUTING.md:12]: WDYT about adding the info for docs contributors?

Comment (kirszenbaum) [reply CONTRIBUTING.md:14]: Some of the other rules (e.g. tests, changelog entry) wouldn't apply to documentation PRs. Maybe add a separate section that redirects to the docs' README, to keep all documentation info in a single place?

d630836

Comment (budnix) [reply CONTRIBUTING.md:14]: Sounds good 👍

**Review by budnix — APPROVED** [review]
👍

**Pull request merged**